presto-product-tests/src/main/java/io/prestosql/tests/hive/TestAvroSymlinkInputFormat.java

```diff
@@ -88,11 +109,51 @@ public void testSymlinkTable()
                 "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat' " +
                 "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'");
 
-        hdfsClient.delete(warehouseDirectory + "/test_avro_schema_symlink/symlink.txt");
-        hdfsClient.saveFile(warehouseDirectory + "/test_avro_symlink/symlink.txt", format("hdfs://%s/TestAvroSymlinkInputFormat/data/original_data.avro", warehouseDirectory));
+        String tableRoot = warehouseDirectory + '/' + table;
+        String dataDir = warehouseDirectory + "/data";
+        String anotherDataDir = warehouseDirectory + "/data2";
+
+        hdfsClient.createDirectory(tableRoot);
+        saveResourceOnHdfs("avro/original_data.avro", dataDir + "/original_data.avro");
+        saveResourceOnHdfs("avro/original_data.avro", anotherDataDir + "/more_data.avro");
+        hdfsClient.saveFile(dataDir + "/dontread.txt", format("This file will cause an error if read as avro."));
+        hdfsClient.saveFile(tableRoot + "/symlink.txt", format("hdfs:%s/original_data.avro", dataDir) + System.lineSeparator() + format("hdfs:%s/more_data.avro", anotherDataDir));
+        assertThat(onPresto().executeQuery("SELECT COUNT(*) as cnt FROM " + table)).containsExactly(row(2));
+        onHive().executeQuery("DROP TABLE " + table);
+        hdfsClient.delete(tableRoot);
+        hdfsClient.delete(dataDir);
+    }
+
+    @Test(groups = {AVRO, STORAGE_FORMATS})
+    public void testSymlinkTableWithDirectory()
```

Comment:
It seems that this test is same as `testSymlinkTable` (nested dir vs non-nested, shouldn't matter).
Add another symlinked file at top-level table directory.

Reply:
This tests when the manifest contains just a directory. It has `hdfsClient.saveFile(tableRoot + "/symlink.txt", format("hdfs://%s/", dataDir));` where `testSymlinkTable` has `hdfsClient.saveFile(tableRoot + "/symlink.txt", format("hdfs:%s/original_data.avro", dataDir));`